### PR TITLE
[exporter/elasticsearch] Add span event to traces OTel mapping mode

### PR DESCRIPTION
**Pull request by carsonip**
**Description:** <Describe what has changed.>

  Span events are now supported in OTel mapping mode.
  They will be routed to `logs-${data_stream.dataset}-${data_stream.namespace}` if `traces_dynamic_index::enabled` is `true`.

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

Updated exporter tests

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Review by crobert-1 — APPROVED** [review]
The logic looks good to me. Since there's no open issue for this that I'm aware of, I'll defer to code owners to confirm this behavior aligns with the goals of this component 👍

File: exporter/elasticsearchexporter/data_stream_router.go
Comment (felixbarny) [inline exporter/elasticsearchexporter/data_stream_router.go:96]: > This function may mutate record attributes.

I wonder if that could be an issue when using multiple exporters. Could this impact the records for other exporters so that there's a race condition where if the ES exporter for an event runs first, the other exporter also sees the mutated values?

I think we mutate records in the ES exporter in other places, too, so this isn't a new issue. If we think it could be a problem, let's create an issue and work towards making the ES exporter not mutate the records.

**Review by felixbarny — APPROVED** [review]

File: exporter/elasticsearchexporter/data_stream_router.go
Comment (carsonip) [reply exporter/elasticsearchexporter/data_stream_router.go:96]: > I think we mutate records in the ES exporter in other places, too, so this isn't a new issue. 

Agreed. I also discovered this potential issue while working on OTel mode. Let me create an issue to track this.

File: exporter/elasticsearchexporter/data_stream_router.go
Comment (andrzej-stencel) [inline exporter/elasticsearchexporter/data_stream_router.go:14]: Why is `nolint:unparam` needed? Seems that all parameters are used?

**Review by andrzej-stencel — APPROVED** [review]

File: exporter/elasticsearchexporter/data_stream_router.go
Comment (carsonip) [reply exporter/elasticsearchexporter/data_stream_router.go:96]: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/35023

**Pull request merged**
